config/confighttp/confighttp.go

```diff
@@ -225,3 +261,9 @@ func (hss *HTTPServerSettings) ToServer(handler http.Handler, settings component
 		Handler: handler,
 	}
 }
+
+func isValidHTTPCompressionKey(compressionType string) bool {
```

Comment:
Not sure what's better: the map, or a switch case here, like:

```go
func isValidHTTPCompressionKey(compressionType CompressionType) bool {
	switch compressionType {
	case CompressionDeflate,
		CompressionGzip,
		CompressionSnappy,
		CompressionZlib,
		CompressionNone,
		CompressionZstd:
		return true
	default:
		return false
	}
}
```

Even better would be if mapstruct would support Enums. Do you know whether they do?

Reply:
changed to switch case